### PR TITLE
test: PocketIC bitcoin integration tests on macOS

### PR DESCRIPTION
**Pull request by mraszyk**
This PR enables PocketIC bitcoin integration tests on macOS (both Intel and arm64).

### CODE REVIEW

**Comment by rumenov** [comment]
Why the existing tests fail on Mac ?

**Comment by mraszyk** [comment]
> Why the existing tests fail on Mac ?

They don't fail, they were just not enabled so far.

File: rs/bitcoin/adapter/BUILD.bazel [hunk 1]
```diff
@@ -109,10 +109,10 @@ rust_test_suite(
         "@bitcoin_adapter_mainnet_headers//file",
         "@bitcoin_adapter_testnet_blocks//file",
         "@bitcoin_adapter_testnet_headers//file",
-        "@bitcoin_core//:bitcoind",
+        "@bitcoin_core_x86_64-linux-gnu//:bitcoind",
     ],
     env = {
-        "BITCOIN_CORE_PATH": "$(rootpath @bitcoin_core//:bitcoind)",
+        "BITCOIN_CORE_PATH": "$(rootpath @bitcoin_core_x86_64-linux-gnu//:bitcoind)",
```
Comment (rumenov) [inline rs/bitcoin/adapter/BUILD.bazel:115]: There must a better way of doing it.  I don't think here we should even mention the platform

cc: @nmattia  @basvandijk

Comment (mraszyk) [reply rs/bitcoin/adapter/BUILD.bazel:115]: https://dfinity.slack.com/archives/CL7Q2RXUM/p1732785051529109

Comment (nmattia) [reply rs/bitcoin/adapter/BUILD.bazel:115]: sure thing! we do this for [jemalloc](https://github.com/dfinity/ic/tree/master/third_party/jemalloc) and [lmdb](https://github.com/dfinity/ic/tree/master/third_party/lmdb), among others. Let me know if you need a hand!

Comment (mraszyk) [reply rs/bitcoin/adapter/BUILD.bazel]: I could do it by reverting #1603 : it was removed according to [this](https://github.com/dfinity/ic/pull/1491#discussion_r1762760330) conversation.

Comment (mraszyk) [reply rs/bitcoin/adapter/BUILD.bazel]: Note that the bitcoind version becomes downgraded by reverting the commit: 27.0 => 25.0. Are we fine with that?

Comment (mraszyk) [reply rs/bitcoin/adapter/BUILD.bazel]: Building from sources doesn't work on macOS: https://github.com/dfinity/ic/actions/runs/12089105211/job/33713847474?pr=2890 - how should we proceed (I'd very much like to have bitcoin tests for PocketIC on macOS, but don't have time to debug building bitcoind from sources on macOS)? Would it be fine to download pre-built artifacts as I did originally after all?

Comment (mraszyk) [reply rs/bitcoin/adapter/BUILD.bazel]: Followed the approach from this PR: https://github.com/dfinity/ic/pull/2996

File: MODULE.bazel [hunk 2]
```diff
@@ -768,7 +768,7 @@ cc_library(
 )
 
 http_archive(
-    name = "bitcoin_core",
+    name = "bitcoin_core_linux",
```
Comment (nmattia) [inline MODULE.bazel:771]:
```suggestion
    name = "bitcoin_core_linux_x86",
```

would suggest adding the arch to keep it consistent with the darwin ones

**Review by nmattia — APPROVED** [review]

**Review by rumenov — APPROVED** [review]

**Review by fxgst — APPROVED** [review]

**Pull request merged**